src/compile/layer.ts

```diff
@@ -107,15 +107,43 @@ export class LayerModel extends Model {
     this.component.layout = parseLayerLayout(this);
   }
 
-  public parseScale() {
+  /**
+   * Convert the domain to an array of data refs. Also, thorw away sorting information
+   * since we always sort the domain when we union two domains.
+   */
+  private static normalizeDomain(domain: DataRefUnionDomain | FieldRefUnionDomain | VgDataRef): VgDataRef[] {
```

Comment:
Somehow this comment was collapsed.  

- FYI, we rarely use static method.  Extracting it as a function on its own would be more consistent + make function call shorter (just `normalizeDomain` rather than `LayerModel.normalizeDomain`.  

- Even if your comment below it true, I suspect that we will still re-use it in concat?

> Not really because this helper does not operate the full type. Arrays are handled differently.